twilio/twilio-cli-core
# fix: Add http agent to axios to work with proxy

Fixes twilio/twilio-cli#223

For the Twilio CLI to work behind an https proxy, we can now set the proxy URL in an env variable called `HTTP_PROXY`. This will set the appropriate `httpAgent` in axios.

### Testing instructions
- Have a locally running forward proxy
- Be on a network behind a https authenticating proxy
- Set environment variable `HTTP_PROXY` to have the proxy's url

On `main`:
- Run any Twilio CLI command
- Make sure the command fails

On this branch:
- Run any Twilio CLI command
- Make sure the command succeeds

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli-core/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added the necessary documentation about the functionality in the appropriate .md file
- [x] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.
